ingest/expression_files/mtx.py

```diff
@@ -24,144 +20,221 @@
-            gene_models.append(
-                self.Model(
-                    {
-                        "name": gene,
-                        "searchable_name": gene.lower(),
-                        "study_file_id": self.study_file_id,
-                        "study_id": self.study_id,
-                        "gene_id": gene_id,
-                        "_id": id,
-                    }
-                )
+        exp_cells = []
+        exp_scores = []
+        visited_expression_idx = [0]
+        model_id = None
+
+        # All observed cells
+        data_arrays.extend(
+            GeneExpression.create_data_array(
+                name=f"{self.cluster_name} Cells",
+                array_type="cells",
+                values=self.cells,
+                linear_data_type="Study",
+                linear_data_id=self.study_file_id,
+                **self.da_kwargs,
             )
-            # Collect all cells and expression scores associated with a gene
-            for row in matched_rows:
-                raw_gene_idx, raw_barcode_idx, raw_exp_score = row.split()
-                exp_cell = self.cells[int(raw_barcode_idx) - 1]
-                exp_score = round(float(raw_exp_score), 3)
-                exp_cells.append(exp_cell)
-                exp_scores.append(exp_score)
-            for cell_data_array in self.set_data_array_gene_cell_names(
-                gene, id, exp_cells
-            ):
-                data_arrays.append(cell_data_array)
-            for gene_data_array in self.set_data_array_gene_expression_values(
-                gene, id, exp_scores
-            ):
-                data_arrays.append(gene_data_array)
-            if len(gene_models) == 5:
-                num_processed += len(gene_models)
-                self.info_logger.info(
-                    f"Processed {num_processed} models, {str(datetime.datetime.now() - start_time)} elapsed"
-                )
-                yield (gene_models, data_arrays)
-                gene_models = []
-                data_arrays = []
-                self.info_logger.info(
-                    f"Processed {num_processed} models, {str(datetime.datetime.now() - start_time)}"
-                )
-        yield gene_models, data_arrays
-        num_processed += len(gene_models)
-        self.info_logger.info(
-            f"Processed {num_processed} models, "
-            f"{str(datetime.datetime.now() - start_time)}"
         )
-
-    @trace
-    def close(self):
-        """
-        Closes file barcode and features file
-        """
-        self.genes_file.close()
-        self.barcodes_file.close()
+        for row in self.mtx_file:
+            raw_gene_idx, raw_barcode_idx, raw_exp_score = row.split()
+            current_idx = int(raw_gene_idx)
+            gene_id, gene = self.genes[current_idx - 1].split("\t")
+            if current_idx != last_idx:
+                is_sorted = MTXIngestor.is_sorted(current_idx, visited_expression_idx)
```

Comment:
See other comment re `is_foo` vs. `check_foo`.

```suggestion
                is_sorted = MTXIngestor.check_sorted(current_idx, visited_expression_idx)
```